immer/extra/persist/README.rst

```diff
@@ -79,8 +79,22 @@ Which generates some JSON like this:
 As you can see, the value is serialized with every ``immer`` container replaced by an identifier.
 This identifier is a key into a pool, which is serialized just after.
 
-- XXX: what's a pool?
-- XXX: to_json_with_pool uses cereal
+A pool is a data structure that ``immer-persist`` uses to work with the underlying structure of the ``immer`` containers.
+For example, ``vector`` and ``flex_vector`` use Radix Balanced Trees internally that have inner nodes and leaf nodes. These
+nodes can be seen in the JSON above. In other words, containers can be put into a pool and later can be retrieved from it, all while preserving
+the structural sharing. One pool can only work with one container type.
```

Comment:
Try to think from the user's point of view, instead of from the implementors point of view. For example, here you say that the pool is a data-structure that is used to work with the underlying structure and then go to talk about nodes. That is kind of true for the implementer, but the user doesn't care so much about nodes and what not.

This is an alternative explanation, that focuses on, from the reader's POV, what the pool is conceptually (what a pool value represents) and then why it's needed (which looks at the implementation but from the POV of why this is useful to the user).

> A pool represents a _set_ of Immer containers of a given type. For example, we may have a pool that contains all `immer::vector<int>` of our document. You can think of it as a little data base of Immer containers. When serializing the pool, the internal structure of all those Immer containers is written, preserving the structural sharing between those containers. The nodes of the trees that implement the Immer containers are represented directly in the JSON and, because we have are representing all the containers as a whole, those nodes that are referenced in multiple trees can be stored only once. That same structure is preserved when reading the pool back from disk and reconstructing the vectors from it, thus allowing us to preserve the structural sharing across sessions.

Reply:
Also maybe worth mentioning the distinction between output and input pool. Unless there is a away to make it so that only one type of pool exists in the code, without much performance penalty? That would make it simpler for the user.